### PR TITLE
ci/contributing: verify donation address/qr's are signed

### PR DESCRIPTION
**Pull request by plowsof**
with this PR i have moved the general fund donation addresses/qr checksums into a _data file that can be signed. this is confirmed in the workflow

thoughts? if this is useful we can ask bF to verify/sign the file.

- [x] Btc QR code with a consistent address
- [x] Confirm _data is gpg signed
- [x] Check hashes of QR's
- [x] Confirm scanned QR content == expected 
- [x] Monero QR now has "monero:" uri 
- [x] Confirm hash of contributing.yml
- [x] Upload signed contributing.yml.asc at a later date

### tldr:
```
git clone https://github.com/plowsof/monero-site.git
cd monero-site
git fetch
git checkout contrib_data
cd _data
gpg --output contributing.yml.asc --armor --detach-sig contributing.yml
```
send contributing.yml.asc  :heavy_check_mark:

### CODE REVIEW

**Comment by nahuhh** [comment]
LGTM. I have no preference over who signs the file, as long as its signed and correct.

if using your key, perhaps we should use your pgp key from the gitian.sigs repo(?).

**Comment by plowsof** [comment]
My plan is to either:
1. Drop / amend the commit adding the signed file with binaryFates signed version.
2. Add the hardcoded checksum in the workflow file and merge asap while we await the signed file (if that causes delays, but i've not asked yet)

i have sanity checked that the workflow fails correctly:
- file has to be signed
- external things such as diffs in checksum of qr images
- the content of the QRs must match exactly what resides in contributing.yml (both xmr/btc) 

after first approval(s) i can proceed with 1 OR 2

**Review by nahuhh — APPROVED** [review]

File: .github/workflows/hashes.yaml [hunk 1]
```diff
@@ -17,13 +19,43 @@ jobs:
...
           gpg --verify downloads/hashes.txt
+          # signed by myself for test purposes only.
+          curl -sL https://raw.githubusercontent.com/plowsof/pgp/refs/heads/main/plowsofsmol.asc |
+            gpg --import
+          gpg --verify _data/contributing.yml.asc
+      - name: Verify General Fund donation QR's
+        run: |
+          yaml="_data/contributing.yml"
+          get_yaml_value() {
+              awk -v key="$1:" '$1 == key {print $2}' "$yaml"
```
Comment (HardenedSteel) [inline .github/workflows/hashes.yaml:37]: I'm not familiar with awk but looks like easy to break things when the content of the site changed a bit. Is it?

Comment (nahuhh) [reply .github/workflows/hashes.yaml:37]: awk is a text processor and $1 processes the first argument $2 the second etc.

an example is
`echo uhh nah | awk '{print $2 $1}'

Comment (plowsof) [reply .github/workflows/hashes.yaml:37]: also, we're only looking at "contributing.yml" . this is a hack to access yaml variables. astro has no problem importing yml files / using their variables so the same signed file will be usable by both.

**Review by HardenedSteel — APPROVED** [review]

**Comment by plowsof** [comment]
Thank you for the reviews. i will ask bF to sign the file for us

**Comment by nahuhh** [comment]
```
-----BEGIN PGP SIGNATURE-----

iHUEABYIAB0WIQTizsOmloPPzjLdLyBrBkSurAmTEAUCZ8skkwAKCRBrBkSurAmT
EMKHAPsGY4AEii8t+HIe6vxMRLNLgLwnh8df7ueg80JiUlxh7wEAxnPUw4Ga1tFl
z7lPQ6RDOFzJuQiK4SChSBvjoMeTUAQ=
=Iwa/
-----END PGP SIGNATURE-----
```

**Comment by plowsof** [comment]
thanks @nahuhh confirmed :
```
gpg:                using EDDSA key E2CEC3A69683CFCE32DD2F206B0644AEAC099310
gpg: Good signature from "nahuhh <nah.uhh@protonmail.com>" [unknown]
```
and my sig in this PR as a dummy file:
```
gpg:                using ECDSA key DC8BCF0C5083F2030782803E4E8A912D40ED052C
gpg: Good signature from "plowsof <plowsof@protonmail.com>" [uncertain]
```

**Review by selsta — APPROVED** [review]

**Comment by binaryFate** [comment]
-----BEGIN PGP SIGNATURE-----

iQIzBAABCAAdFiEErVZM2o8WZazni139JZODjqux9lUFAmfRkmUACgkQJZODjqux
9lWqxhAAmvcCht95StPoeOqwWvZd1Y47HfJSezH7/1VFOKlIJFXGmKZbXVdlCRgd
KXHXgCGvFsEBC7tt5tEPLQ/u4O4B5vflQ/s2VzBeen7vHMHOrM2WyU4ycYE8Y5yz
3SIJaDIx7dZUZrk/8OdNQientZdRuT4xfkRTwUntIODc+u3AdAhiYpqrMUyp3SZo
+0DZ1OVuYbZG13MwyeWiSnBc4b7DTiAzGbILyU3gpAhmEB4/hIKlvrrf5RtrOgdm
EPwwkhwx9JQqhfHO927i0gpSwk3WnTJDmKwZBTuOWnjKG5yblBf5JpPs/AjO1ZQV
ZGB2wkcf2eWrpWbUdQj/il17Fyr9z3zYCQTrwAySLfDyClzi0VAKN9Igu4OO+hYY
5oaKVNKvuC/0kksEeRWu/WkD5cmhEHuAMn1ytvI7udfW9Bc7mfHX5LyFDoInGvzW
wcoB9ulunShj94ctUwKJ+PvUhoMvwDNxJn/awoUyvzDELtxk+pF03oOS9oMpQWC4
vo7y0toSN6KtXyDETWGozFx4ph+eM4JvsFO2SGTRLgv+T+izTR1C+iQ/JNpicDlS
20Z5BhZaFitdK8+dOCifDqDuTsZ23M8C5k0Pp/7mw9f/bvAwLQ/PQTfgIqqfUA2E
B8arkPOgg7uIJP+Go2H/2brhrFHE60Sh8K2i6pBhOiHS6ChCYOM=
=FrXY
-----END PGP SIGNATURE-----

**Review by nahuhh — APPROVED** [review]
